script.js

```diff
@@ -1,4 +1,181 @@
+var reverseModeStatus = false;
+var userWinCount = 0;
+var computerWinCount = 0;
+var drawCount = 0;
+var userName = "";
+
+var RandomSPS = function () {
+  randomDecimal = Math.random();
+  randomInteger0to2 = Math.floor(randomDecimal * 3);
+  RandomInterger1to3 = randomInteger0to2 + 1;
+  if (RandomInterger1to3 == 1) {
+    computerChoiceMsg = "The computer chose Scissors!";
+    return "Scissors";
+  }
+  if (RandomInterger1to3 == 2) {
+    computerChoiceMsg = "The computer chose Paper!";
+    return "Paper";
+  }
+  if (RandomInterger1to3 == 3) {
+    computerChoiceMsg = "The computer chose Stone!";
+    return "Stone";
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  SPS = RandomSPS();
+
+  while (userName == "") {
+    userName = input;
+    if (userName == "") {
+      myOutputValue = "Please enter a valid username and click submit.";
+      return myOutputValue;
+    }
+    myOutputValue =
+      "Hi " +
+      userName +
+      ", play the game by entering either Scissors, Paper, or Stone and click submit!";
+    return myOutputValue;
+  }
+
+  if (input == "Reverse") {
+    reverseModeStatus = true;
+    return "Secret Reverse Mode On!";
+  }
+
+  if (input == "Reverse Off") {
+    reverseModeStatus = false;
+    return "Secret Reverse Mode Off!";
+  }
+
+  if (
+    input != "Scissors" &&
+    input != "Paper" &&
+    input != "Stone" &&
+    input != "Reverse" &&
+    input != "Reverse Off"
+  ) {
+    yourChoiceMsg = "Please input either Scissors, Paper, or Stone.";
+    result = "You did not pick a valid option.";
+  }
+
+  if (!reverseModeStatus) {
+    if (SPS == "Scissors") {
+      if (input == "Scissors") {
+        drawCount = drawCount + 1;
+        yourChoiceMsg = "You chose Scissors!";
+        result = "It is a draw!";
+      }
+      if (input == "Paper") {
+        computerWinCount = computerWinCount + 1;
+        yourChoiceMsg = "You chose Paper!";
+        result = "You lose!";
+      }
+      if (input == "Stone") {
+        userWinCount = userWinCount + 1;
+        yourChoiceMsg = "You chose Stone!";
+        result = "You win!";
+      }
+    }
```

Comment:
Good job mastering ifs !  

We'll be learning if else soon which will make this sort of logic easier :)